packages/@react-stately/virtualizer/src/LayoutInfo.ts

```diff
@@ -94,6 +94,7 @@ export class LayoutInfo {
     res.transform = this.transform;
     res.parentKey = this.parentKey;
     res.isSticky = this.isSticky;
+    res.zIndex = this.zIndex;
```

Comment:
Virtualizer calls `layoutInfo.copy()` when updating the item size. Previously this meant the `zIndex` calculated within TableLayout would get reset to 0 since `copy` wouldn't carry over the `zIndex` causing the original bug. Only happened for `overflowMode: wrap` because the rowHeight is estimated

Reply:
Great catch!